File: 001_starting_MATLAB/solution/det2x2.m
function d = det2x2(A)
%DET2X2 Compute the determinant of a 2x2 matrix
%
% d = det2x2(S);
%
% Inputs:
%  A  a 2x2 matrix
% Outputs:
%  d  the determinant of A

d = A(1,1)*A(2,2) - A(1,2)*A(2,1);

end

File: 002_NumPy/solution/det2x2.py
import numpy as np
def det2x2(A):
    """
    Compute the determinant of a 2x2 matrix.

    This function returns the determinant of the 2x2 matrix A.
    """
    return A[0,0]*A[1,1] - A[0,1]*A[1,0]
